src/DmaRequestCore.bsv

```diff
@@ -22,78 +24,80 @@ module mkChunkComputer(ChunkCompute ifc);
 
     Reg#(DmaMemAddr) newChunkPtrReg <- mkReg(0);
     Reg#(DmaMemAddr) totalLenRemainReg <- mkReg(0);
-    Reg#(Bool) isSplittigReg <- mkReg(False);
+    Reg#(Bool) isSplittingReg <- mkReg(False);
 
     function Bool hasBoundary(DmaRequestFrame request);
-        let highIdx = (request.startAddr + request.length) >> BUS_BOUNDARY_WIDTH;
-        let lowIdx = request.startAddr >> BUS_BOUNDARY_WIDTH;
+        let highIdx = (request.startAddr + request.length - 1) >> valueOf(BUS_BOUNDARY_WIDTH);
+        let lowIdx = request.startAddr >> valueOf(BUS_BOUNDARY_WIDTH);
         return (highIdx > lowIdx);
     endfunction
 
     function DmaMemAddr getOffset(DmaRequestFrame request);
-        DmaMemAddr offset = zeroExtend(fromInteger(valueOf(BUS_BOUNDARY)) - pack(request.startAddr[BUS_BOUNDARY_WIDTH-1:0]));
-        return offset;
+        // 4096 - startAddr % 4096
+        Bit#(BUS_BOUNDARY_WIDTH) remainder = truncate(request.startAddr);
```

Comment:
`Bit#(BUS_BOUNDARY_WIDTH)` has occured more than once, so use `typedef` to define a new type. 
Duplicated code is a kind of bad smell in your codebase